src/components/common/TokenIcon/index.tsx

```diff
@@ -108,29 +114,64 @@ const currenciesData = [
 ]
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const getTokenData = (symbol: string): Array<any> => {
-  return currenciesData.filter((item) => item.symbol.toUpperCase() === symbol.toUpperCase())
+const getTokenData = (symbol: string): Maybe<CurrencyData> => {
+  const currenciesDataFiltered = currenciesData.filter(
+    (item) => item.symbol.toUpperCase() === symbol.toUpperCase()
+  )
+  return currenciesDataFiltered.length > 0 ? currenciesDataFiltered[0] : null
 }
 
 interface Props {
-  symbol: string
+  token: Token
   onClick?: () => void
 }
 
 export const TokenIcon: React.FC<Props> = (props) => {
-  const { onClick, symbol, ...restProps } = props
-  const data = getTokenData(symbol)
+  const { onClick, token, ...restProps } = props
+  const { address, symbol } = token
+  const [isTokenChecked, setIsTokenChecked] = React.useState<Maybe<boolean>>(null)
+  const [isTokenImageExist, setIsTokenImageExist] = React.useState<boolean>(false)
+
+  const currencyData = getTokenData(symbol)
+
+  // Only exist for the mainnet, not for rinkeby
+  const customImageUrl = `https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/${address}/logo.png`
+
+  React.useEffect(() => {
+    let cancelled = false
+
+    const CheckTokenAvailability = async () => {
+      try {
+        const result = await fetch(customImageUrl, { method: 'HEAD' })
+        if (!cancelled) setIsTokenImageExist(result.ok)
+      } catch (err) {
+        if (!cancelled) setIsTokenImageExist(false)
+      }
+      if (!cancelled) setIsTokenChecked(true)
+    }
+    CheckTokenAvailability()
+
+    return () => {
+      cancelled = true
+    }
+  }, [customImageUrl])
```

Comment:
In my opinion <TokenIcon> should be a presentational component. Maybe currencyData type could be extended to have an optional custom URL an fetch everything for a list of that type and pass options to TokenIcon. 

If we maintain this logic, I think we can move customImageUrl to a function to have it extracted in case we need to use another URL for rinkeby, or use this functionality inside another component or hook. The effect could be move to a hook. Something like `const { customUrl, tokenChecked, tokenImageExist } = useCustomUrl(address)`

Reply:
Yes, let's go with this